packages/auth/src/saml/saml-flow.ts

```diff
@@ -33,18 +33,21 @@ export function buildSamlFlow(config: SamlFlowOptions): SamlFlow {
     },
 
     async exchangeHandshakeToken() {
-      const {location, history, request} = options;
+      const {organizationId, location, history, request} = options;
       const handshakeToken = getHandshakeToken(location);
       removeHandshakeToken(location, history);
 
       try {
-        const response = await request(`${api}/handshake/token`, {
-          method: 'POST',
-          body: JSON.stringify({handshakeToken}),
-          headers: {
-            'content-type': 'application/json; charset=UTF-8',
-          },
-        });
+        const response = await request(
+          `${api}/handshake/token?organizationId=${organizationId}`,
```

Comment:
You could `encodeURIComponent` the orgId for good measure.

However I think all org ID are url safe characters anyway, so that's kind of a belt and suspenders situation 🤷 

It's more a reflex at this point hehe.